Assets/Scripts/PlayerSpawnTrigger.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerSpawnTrigger : MonoBehaviour
+{
+	private Transform playerRespawnTransform;
+	// Start is called before the first frame update
+	void Start()
+	{
+		playerRespawnTransform = GameObject.Find("Game Manager").transform.Find("PlayerRespawnPoint").transform;
+	}
+
+	void OnTriggerEnter(Collider collider)
+	{
+		//If the object is the player who shot
+		if (collider.gameObject.tag == "Player")
+		{
```

Comment:
Let's try to use `gameObject.CompareTag("tag")` in these situations, since it's more recommended and more efficient as it says